packages/yourloops/test/integration/assert/patient-list.ts

```diff
@@ -580,3 +586,45 @@ export const checkPatientListTooltips = async (): Promise<void> => {
     expect(screen.queryByRole('tooltip')).not.toBeInTheDocument()
   })
 }
+
+export const checkPatientListColumnSort = async (): Promise<void> => {
+  const dataGridRows = screen.getByTestId('patient-list-grid')
+
+  const patientColumnHeader = within(dataGridRows).getByRole('columnheader', { name: 'Patient' })
+  const allRowsBeforeSort = within(dataGridRows).getAllByRole('row')
+  expect(allRowsBeforeSort[1]).toHaveTextContent(monitoredPatient.profile.fullName)
+  expect(allRowsBeforeSort[2]).toHaveTextContent(monitoredPatientTwo.profile.fullName)
+  expect(allRowsBeforeSort[3]).toHaveTextContent(monitoredPatientWithMmol.profile.fullName)
+  expect(allRowsBeforeSort[4]).toHaveTextContent(unmonitoredPatient.profile.fullName)
+
+  const sortButton = within(patientColumnHeader).getByRole('button', { hidden: true })
+  await userEvent.click(sortButton)
+
+  const allRowsAfterFirstSort = within(dataGridRows).getAllByRole('row')
+  expect(allRowsAfterFirstSort[1]).toHaveTextContent(unmonitoredPatient.profile.fullName)
+  expect(allRowsAfterFirstSort[2]).toHaveTextContent(monitoredPatientWithMmol.profile.fullName)
+  expect(allRowsAfterFirstSort[3]).toHaveTextContent(monitoredPatientTwo.profile.fullName)
+  expect(allRowsAfterFirstSort[4]).toHaveTextContent(monitoredPatient.profile.fullName)
+
+  await userEvent.click(sortButton)
+
+  const allRowsAfterSecondSort = within(dataGridRows).getAllByRole('row')
+  expect(allRowsAfterSecondSort[1]).toHaveTextContent(monitoredPatient.profile.fullName)
+  expect(allRowsAfterSecondSort[2]).toHaveTextContent(monitoredPatientTwo.profile.fullName)
+  expect(allRowsAfterSecondSort[3]).toHaveTextContent(monitoredPatientWithMmol.profile.fullName)
+  expect(allRowsAfterSecondSort[4]).toHaveTextContent(unmonitoredPatient.profile.fullName)
+}
+
+export const checkMonitoringAlertsIconsInactiveForFirstPatient = async (): Promise<void> => {
```

Comment:
Could it be more generic ? Why do you have to test it on the first patient ? Can we have a test on any patient and not particulary the first one of the list ?

Reply:
I just want to point out that in most cases, this is not helpful to have generic code in test :wink: